dspace-api/src/main/java/org/dspace/versioning/VersioningConsumer.java

```diff
@@ -49,35 +77,428 @@ public void finish(Context ctx) throws Exception {
+        ));
+
+        // reindex right item
+        Item rightItem = relationship.getRightItem();
+        itemsToProcess.add(rightItem);
+        ctx.addEvent(new Event(
+            Event.MODIFY, rightItem.getType(), rightItem.getID(), null, itemService.getIdentifiers(ctx, rightItem)
+        ));
+    }
+
+    /**
+     * Given two items, check if their entity types match.
+     * If one or both items don't have an entity type, comparing is pointless and this method will return false.
+     * @param latestItem the item that represents the most recent version.
+     * @param previousItem the item that represents the second-most recent version.
+     * @return true if the entity types of both items are non-null and equal, false otherwise.
+     */
+    protected boolean doEntityTypesMatch(Item latestItem, Item previousItem) {
+        String latestItemEntityType = getEntityType(latestItem);
+        String previousItemEntityType = getEntityType(previousItem);
+
+        // check if both items have an entity type
+        if (latestItemEntityType == null || previousItemEntityType == null) {
+            if (previousItemEntityType != null) {
+                log.warn(String.format(
+                    "Inconsistency: Item with uuid %s, handle %s has NO entity type, " +
+                    "but the previous version of that item with uuid %s, handle %s has entity type %s",
+                    latestItem.getID(), latestItem.getHandle(),
+                    previousItem.getID(), previousItem.getHandle(), previousItemEntityType
+                ));
             }
+
+            // one or both items do not have an entity type, so comparing is pointless
+            return false;
         }
+
+        // check if the entity types are equal
+        if (!StringUtils.equals(latestItemEntityType, previousItemEntityType)) {
+            log.warn(String.format(
+                "Inconsistency: Item with uuid %s, handle %s has entity type %s, " +
+                "but the previous version of that item with uuid %s, handle %s has entity type %s",
+                latestItem.getID(), latestItem.getHandle(), latestItemEntityType,
+                previousItem.getID(), previousItem.getHandle(), previousItemEntityType
+            ));
+            return false;
+        }
+
+        // success - the entity types of both items are non-null and equal
+        log.info(String.format(
+            "Item with uuid %s, handle %s and the previous version of that item with uuid %s, handle %s " +
+            "have the same entity type: %s",
+            latestItem.getID(), latestItem.getHandle(), previousItem.getID(), previousItem.getHandle(),
+            latestItemEntityType
+        ));
+        return true;
+    }
+
+    /**
+     * Get the entity type (stored in metadata field dspace.entity.type) of any item.
+     * @param item the item.
+     * @return the label of the entity type.
+     */
+    protected String getEntityType(Item item) {
```

Comment:
The `RelationshipMetadataService.getEntityTypeFromMetadata(context,item)` returns an `EntityType` object and requires a context (currently not present in all calls to this method), but if you'd prefer we can check whether it can be refactored

Reply:
@benbosman : Honestly, I'll leave it up to you which is easier.  I just noticed the same logic duplicated in both areas.  If it's easy enough to create/pass a Context to this method, then we may want to refactor (to avoid duplicate code).  If that's too complex though, I'm OK with this minor duplication for now.